src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -310,6 +321,42 @@ def checkm2(genome_ids, work_dir, source_data_dir, debug, program_threads, batch
     return failed_ids
 
 
+def microtrait(genome_ids, work_dir, source_data_dir, debug, program_threads, batch_number, node_id, source_file_ext):
+    failed_ids, size = list(), len(genome_ids)
+    print(f'Start executing MicroTrait for {len(genome_ids)} genomes')
+
+    batch_dir = _create_batch_dir(work_dir, batch_number, size, node_id)
+    tool_genome_id_map, source_genome_file_map = dict(), dict()
+    # retrieve genomic.fna.gz files
+    fna_files = list()
+    for genome_id in genome_ids:
+        genome_file = _find_genome_file(genome_id, source_file_ext, source_data_dir,
+                                        exclude_file_name_substr=['cds_from', 'rna_from', 'ERR'],
+                                        expected_file_count=1)
+
+        if genome_file:
+            tool_genome_id_map.update(_map_tool_id_to_genome_id('microtrait',
+                                                                genome_id,
+                                                                os.path.basename(genome_file[0]),
+                                                                source_file_ext))
+            source_genome_file_map[genome_id] = genome_file[0]
+            fna_files.append(str(genome_file[0]))
+
+    start = time.time()
+
+    # RUN MicroTrait in serial
+    args_list = [(*genome_file, debug) for genome_file in list(source_genome_file_map.items())]
+    results = [_run_microtrait(*args) for args in args_list]
```

Comment:
I was attempting to implement parallelization but found a few issues. 

- I wasn't able to use threads due to rpy2 threads are blocking each other. In my case it's even worse. My second thread won't be able to import `microtrait` due to GIL. (https://stackoverflow.com/questions/48952411/how-can-rpy2-r-code-from-python-be-run-in-a-thread-without-blocking-other-thre)
- Using process here is also problematic. In `main`, we already split each tool operation call into multiple process. We need to turn daemon off in order to support children processes within main parent processes. I think it's a bad idea to turn daemon off and since python 3.11, turning off daemon is no longer an option.   

I think it leaves us a few options, 

- Try to implement microtrait child processes with python 3.10
- Run microtrait in serial in compute_attri script. In the taskfarmer script, we made batch size for microtrait small (100 per batch) and run more taskfarmer tasks. 
- Try to implement parallelization with R. This option is the most efficient one in term of performance. But may add maintenance work in the future.

Reply:
My first thought is to put microtrait in `SERIES_TOOLS` and then run one `subprocess` calling the r script per genome id (possibly with a queue if necessary). Then the r script is isolated and you don't have to worry about r2py.

Reply:
I def don't want to be stuck on py3.10 forever